src/args.rs

```diff
@@ -170,6 +174,39 @@ fn parse_auth(src: &str) -> Result<auth::RequiredAuth, ContextualError> {
     })
 }
 
+/// A own header modified from [httparse](https://docs.rs/httparse/1.3.5/src/httparse/lib.rs.html#415-425)
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct Header {
+    /// The name portion of a header.
+    ///
+    /// A header name must be valid ASCII-US, so it's safe to store as a `String`.
+    pub name: String,
+    /// The value portion of a header.
+    ///
+    /// While headers **should** be ASCII-US, the specification allows for
+    /// values that may not be, and so the value is stored as bytes.
+    pub value: Vec<u8>,
+}
+
+impl Header {
+    fn new(header: &httparse::Header) -> Self {
+        Header {
+            name: header.name.to_string(),
+            value: header.value.to_vec(),
+        }
+    }
+}
+
+fn parse_header(src: &str) -> Result<Header, httparse::Error> {
```

Comment:
Is it perhaps possible to directly output a `Result<DefaultHeaders, httparse::Error>` here? It might make everything a bit more ergonomic.

Reply:
Because DefaultHeaders is not Send, it won't let us move to another thread. Is it another way that I could fix this?

Reply:
Can't we clone our way out of this? It's not like performance is very important during arg parsing. In the worst case, we could use the nuclear option: `Arc<RefCell<Header>>`.